src/OnboardingSPA/pages/Steps/DesignHomepageMenu/index.js

```diff
@@ -85,6 +85,14 @@ const StepDesignHomepageMenu = () => {
 
 		for ( const key in homepagesList ) {
 			const homepagePatterns = homepagesList[ key ];
+			// update the header menu pattern if already selected
+			if (
+				currentData.data.partHeader ||
+				currentData.data.partHeader !== ''
+			) {
+				homepagePatterns[ 0 ] = currentData.data.partHeader;
+			}
```

Comment:
Cant remove this code since the individual homepage previews are constructed here and need to replace the selected header menu slug for those individual previews